src/mlpack/methods/cf/cf.hpp

```diff
@@ -258,22 +258,24 @@ class CF
   //! Cleaned data matrix.
   arma::sp_mat cleanedData;
 
-  /**
-   * Helper function to insert a point into the recommendation matrices.
-   *
-   * @param queryIndex Index of point whose recommendations we are inserting
-   *     into.
-   * @param pos Position in list to insert into.
-   * @param neighbor Index of item being inserted as a recommendation.
-   * @param value Value of recommendation.
-   */
-  void InsertNeighbor(const size_t queryIndex,
-                      const size_t pos,
-                      const size_t neighbor,
-                      const double value,
-                      arma::Mat<size_t>& recommendations,
-                      arma::mat& values) const;
-
+  //! Candidate represents a possible recommendation.
+  struct Candidate
+  {
+    //! Value of this recommendation.
+    double value;
+    //! Item of this recommendation.
+    size_t item;
+    //! Trivial constructor.
+    Candidate(double value, size_t item) :
+        value(value),
+        item(item)
+    {};
+    //! Compare the value of two candidates.
+    friend bool operator>(const Candidate& l, const Candidate& r)
+    {
+      return l.value > r.value;
+    };
+  };
```

Comment:
Thanks for sharing that information. I can see you found a real difference in that issue!
Yeah, I use the `Compare` template parameter when deciding between a max heap and min heap.
So, what I don't understand is:  Do you think we can have a real difference between using std::pair with lambda comparison,  or using the Candidate struct with the custom operator< ? Even if the code seems to be the same? Is there any special optimization for std::pair?